File: Stack/Monotonic Stack/Minimum Operations to Convert All Elements to Zero/brute_force.cpp
#include<bits/stdc++.h>
using namespace std;

//Time complextiy :- O(n*d)  , where d is the number of elements in array 
// Space complexity :- O(n)

// Approach :-  
// store the unique element into the set 
// for each number traverse on array and find the minimum number of operations you'll apply

// Link :- https://leetcode.com/problems/minimum-operations-to-convert-all-elements-to-zero/

class Solution {
public:
    int minOperations(vector<int>& nums) {
        // add every element to hash-set
        unordered_set<int> st(nums.begin(), nums.end());
        int n = nums.size(), res = 0;

        for(auto target: st){
            if(target == 0)
                continue;

            bool flag = false;

            for(int i=0; i<n; ++i){
                if(nums[i] == target){
                    if(!flag){
                        flag = true;
                        res++;
                    }
                }
                else if(nums[i] < target)
                    flag = false;
            }
        }
        return res;
    }
};

File: Tree/Binary Search Tree/Binary Searchable Count/brute_force.java
import java.util.*;

// Time complextiy :- O(n*log(n))
// Space complexity :- O(1)

// Approach :-   
// for every element in the array apply binary search and if we find it then it's binary searchable otherwise not 

//Link :- https://www.geeksforgeeks.org/problems/binary-searchable-elements/1

class Solution {
    private boolean binarySearch(int[] arr, int n, int target){
        int low = 0;
        int high = n-1;
        
        while(low <= high){
            int mid = low + (high - low)/2;
            
            if(arr[mid] == target){
                return true;
            }
            else if(arr[mid] > target){
                high = mid-1;
            }
            else{
                low = mid+1;
            }
        }
        return false;
    }
    
    public int binarySearchable(int[] arr) {
        int n = arr.length;
        int res = 0;
        
        for(int i=0; i<n; ++i){
            if(binarySearch(arr, n, arr[i])){
                res += 1;
            }
        }
        return res;
    }
};